subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,113 @@
+cdata = configuration_data()
+
+# Set SIMD config data and flags
+simd_type = 'None'
+if host_cpu_family.startswith('x86')
+  cdata.set('USE_SSE', true)
+  cdata.set('USE_SSE2', true)
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  simd_type = 'SSE2'
+
+elif host_cpu_family == 'arm'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  simd_type = 'NEON (vfpv4)'
+
+elif host_cpu_family == 'aarch64'
+  cdata.set('USE_NEON', true)
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  simd_type = 'NEON (armv8)'
+endif
+
+# Set numerical representation config data
+numerical_type = ''
+fft_library = ''
+if get_option('use-fixed-point')
+  # the fixed-point implementation doesn't use SSE intrinsics (and arch.h
+  # checks for this), however we still let the compiler pack floats into
+  # SIMD calls instructions whenever its cost model permits. 
+  cdata.set('USE_SSE', false)
+  cdata.set('FIXED_POINT', true)
+
+  numerical_type = 'fixed-point'
+  cdata.set('USE_KISS_FFT', true)
+  fft_library = 'Kiss FFT (built-in)'
+
+else
+  cdata.set('FLOATING_POINT', true)
+  numerical_type = 'floating-point'
+
+  cdata.set('USE_SMALLFT', true)
+  fft_library = 'OggVorbis FFT (built-in)'
+endif
+
+# Set C99 variable-length array config data
+cdata.set('VAR_ARRAYS', cc.get_id() != 'msvc')
+
+# Set EXPORT config data
+if cc.get_id() == 'msvc' and get_option('default_library') != 'static'
+  cdata.set('EXPORT', '')
+elif cc.has_function_attribute('visibility')
+  cdata.set('EXPORT', 'extern __attribute__ ((visibility ("default")))')
+else
+  cdata.set('EXPORT', 'extern')
+endif
+
+# Populate config.h from the above data
+configure_file(input : 'config.h.meson',
+               output : 'config.h',
+               configuration : cdata)
+
+# Add accumulated flags to the build
+foreach arg : c_args
+  add_project_arguments(arg, language: 'c')
+endforeach
+
+# Set header include paths
+incdir = include_directories(
+           '.',
+           'include',
+           'include/speex')
+
+# Nested Meson builds
+subdir('include/speex')
+subdir('libspeexdsp')
+
+if get_option('with-example-programs')
+  subdir('examples')
+endif
+
+summary('SIMD instructions', simd_type,                           section : 'Implementation Summary')
+summary('Numerical type',    numerical_type,                      section : 'Implementation Summary')
+summary('FFT library',       fft_library,                         section : 'Implementation Summary')
+summary('Build type',        get_option('buildtype'),             section : 'Build Summary')
+summary('Link time optimization', get_option('b_lto'),            section : 'Build Summary')
+summary('Library type',      get_option('default_library'),       section : 'Build Summary')
+summary('Install prefix',    get_option('prefix'),                section : 'Build Summary')
+summary('Example programs',  get_option('with-example-programs'), section : 'Build Summary')
```

Comment:
Are these particularly interesting in the summary?

Reply:
Yes, we often see copy-and-paste console logs, and seeing these (specifically the first three) help a lot. 
For example, if a library is shared and the user's reporting unresolved .so problems, then they probably moved their binary (so this is a big help). 

We often have users reporting stuttering or frame hitching - and we need to drill in to realize they have some dependencies not optimized; seeing that up-front here is really helpful.

For informed users, having that little status block showing them that their build is optimized (or not), or that they're building shared libs instead of static is all the help they need to dive into the Meson docs and learn those options.

It's important that the summary is provided at the end of the setup phase instead of scolled off their console (some users are on non-GUI framebuffers, and scroll back doesn't even work).

Reply:
I am particularly wondering about the first 3. Build type isn't really interesting compared to the debug/optimization options, I generally assume that shared/static isn't very interesting except to people thinking about distribution, and lto, particularly, is a useless option because the subproject values are ignored and only the superproject value matters, so... there's nothing to print here.

Reply:
Knocked out the entire Build Summary. Default config is now:

``` python
        default_options : [
          'c_std=c99',
          'default_library=both',
          'warning_level=0'])
```

Summary is just build details:

``` text
  Library Summary
    Optimization level: 0
    SIMD instructions : SSE2
    Numerical type    : floating-point
    FFT library       : OggVorbis FFT (built-in)
```

Reply:
Knocked out the entire Build Summary, as suggested.  Remaining summary is as follows:

``` text
  Library Summary
    Optimization level: 0
    SIMD instructions : SSE2
    Numerical type    : floating-point
    FFT library       : OggVorbis FFT (built-in)
```